src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs

```diff
@@ -5,14 +5,14 @@
 
 namespace Microsoft.PowerPlatform.PowerApps.Persistence.PaYaml.Models.SchemaV3;
 
-public enum DataSourceInstanceType
+public enum DataSourceType
 {
-    DataverseTable,
+    Table,
+    Actions
 }
-
 public record DataSourceInstance
 {
     [YamlMember(DefaultValuesHandling = DefaultValuesHandling.Preserve)]
-    public required DataSourceInstanceType Type { get; init; }
-    public string? TableLogicalName { get; init; }
+    public required DataSourceType Type { get; init; }
```

Comment:
this OM doesn't match the design in the description.
Namely, the `ConnectorId` property is missing. Also, it should be required. Do not create a special case for Dataverse.

Reply:
I take back it needing to be required.

Reply:
If ConnectorId is only required for non-dataverse datasources. If it's is missing, the default value will be dataverse.